File: FlutterC2W/46-AsynchronousProg/prog1.java
class ThreadDemo {
    public static void main(String[] args) {
        // Single thread in java can't run asynchronously while dart having isolates like functionality
        System.out.println(Thread.currentThread().getName());

        System.out.println("Statement1");
        System.out.println("Statement2");
        try {
            Thread.sleep(5000);
        } catch(InterruptedException ie) {
            System.out.println(ie);
        }
        System.out.println("Statement3");
        System.out.println("Statement4");
    }
}

File: FlutterC2W/46-AsynchronousProg/prog1.dart
String getName() {
  return "HelloFlutter";
}
void main() {
  print("statement1");
  print("statement2");
  // Future.delayed(Duration(seconds: 5), ()=>print("statement3"));
  Future.delayed(Duration(seconds: 5), ()=>print(getName()));
  print("statement4");
  print("statement5");
}